Mappers/HttpMapper.php

```diff
@@ -1,226 +1,237 @@
+        return true;
     }
 
-    /**
-     * @deprecated
-     * @param $action
-     * @param $url
-     * @param $options
-     * @return ArraySerializable|\Modules\LibraryModule\Models\Collection|void
-     * @throws EnergonException
-     */
-    protected function sendHttpRequestForGetStatus($action, $url, $options)
-    {
-        return $this->getHttpClient()->sendHttpRequestForGetStatus($action, $url, $options);
-    }
 
     /**
-     * @param ArraySerializable $object
-     * @param $identifierName
-     * @param $urlName
-     * @return string
+     * @param array $objectData
+     * @return Collection|mixed
+     * @throws InternalException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    protected function getUrlWithObjectAndIdentifier(ArraySerializable $object, $identifierName, $urlName)
+    public function updatePatch(array $objectData)
     {
-        $identifier = $object->toArray()[$identifierName];
-        return vsprintf($this->getHttpMapperConfig($urlName), [$identifier]);
+        return $this->getModel(
+            $this->requestFactory->create(self::PATH, $this->urlRender->prepareUpdateUrl($objectData), [], $objectData)
+        );
     }
 
     /**
-     * @param $identifierValue
-     * @param $url
-     * @return string
+     * @param RequestInterface $request
+     * @return \Infrastructure\Models\Http\ResponseInterface
+     * @throws InternalException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    private function getUrlWithIdentifier($identifierValue, $url)
+    protected function sendRequest(RequestInterface $request)
     {
-        $urlTemplate = $this->getHttpMapperConfig($url);
-        return vsprintf($urlTemplate, [$identifierValue]);
+        return $this->getHttpClient()->send($this->mergeDefaultData($request));
     }
 
     /**
-     * @return HttpClient
+     * @param RequestInterface $request
+     * @return mixed
+     * @throws InternalException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    protected function getHttpClient()
+    protected function getModel(RequestInterface $request): ArraySerializable
     {
-        return $this->httpClient;
+        return $this->buildObject($this->sendRequest($request)->getParsedBody());
     }
 
     /**
-     * @param HttpClient $httpClient
-     * @return HttpMapper
+     * @param RequestInterface $request
+     * @return Collection
+     * @throws InternalException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    private function setHttpClient($httpClient)
+    protected function getCollection(RequestInterface $request): Collection
```

Comment:
Change please methods naming. SendRequestForCollection to be more clear.

Reply:
change name